src/app/components/generics/data-list/data-list.component.ts

```diff
@@ -15,84 +8,134 @@ import { SelectionModel } from '@angular/cdk/collections';
 })
 export class DataListComponent implements OnInit {
   @Input() data: any[];
+  @Input() variableNames: string[];
   @Input() dataColumns: string[];
-  @Input() allowSelection: boolean;
+  @Input() allowAdd: boolean;
+  @Input() allowDelete: boolean;
   @Input() allowSearch: boolean;
-  @Input() variableNames: string[];
+  @Input() allowSort: boolean;
```

Comment:
why don't you set default values? This shortens the HTML tag

Reply:
If you explicitly mean the boolean, it has a default = false, so if you do not want the sorting functionalty you can omit it from the HTML tag in the parent. For other optional values I will have to check

Reply:
Or we enable stuff by default and allow the parent to disable it if not needed

Reply:
nah you are right, doesn't really make sense